### PR TITLE
4/NPC-Atlas-Update: Optimised NPC Atlases and Added Selection Highlight Feature

### LINKED ISSUE #396: Atlas' Optimisations
# **Description**
**Task:** [Atlas' Optimisations](#396).
**Feature:** [[Team 4] Friendly NPC Interaction Update and Extensions](#326)

The task is to combine all friendly NPC (Non-Player Character) atlas files into a single image to optimize loading times and improve game performance. Additionally, the Turtle, Lion, and Snake character artworks must be updated to align with the overall game aesthetics. An outline/glowing effect will also be added to the FNPC's to allow for visual identification during player-NPC interaction (i.e. the NPC the player is talking to will glow).
 
# **Dependencies**
There are no dependencies as this is just updates and optimisations. 
 
# **Milestones**
- [x] Create new Turtle, Lion and Snake artwork (17 Sep).
- [x] Combine all FNPCs into one PNG (17 Sep).
- [x] Add outline/glowing effect to the FNPC frames (19 Sep).
- [x] Update atlas code for each NPC (19 Sep).
- [ ] Update Art and atlas related documentation (21 Sep).
 
**Completion Deadline:** Sep. 22

# **Documentation**
- [Main description of feature](../wiki/Game-Screens)
- [JavaDoc](JavaDoc/Location/Package/Class)
- ...
 
# **Member**
- Nahin (Reyana) (@nreyana )

### PR DESCRIPTION
**Pull request by nreyana**
# Description

- Combined all the Friendly NPC's into one png to reduce initial loading time. 
- Updated the atlases accordingly and changed the art for the Turtle and Lion to match the game's aesthetic. 
- Added a "selected" version of the sprites, where there is a white outline/border around the animals that appear when the player interacts with the NPC. This was done to allow for easy identification of which animal is currently active. 

Closes #396 and #397 

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Visual testing was deemed as the easiest and most feasible method to ensure the selection feature is working properly. You can simply approach a friendly NPC, and the outline will appear - this happens due to the "selected" sections in the atlas being using instead of the float option. When multiple NPCs are near the player, they are also selected. This allows for visual clarity to help the player confirm which animal they are interacting with.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/services/DialogueBoxService.java [hunk 1]
```diff
@@ -67,31 +70,37 @@ public void hideCurrentOverlay() {
...
      */
     public void updateText(String[][] text) {
         hints = text;
-        if (currentOverlay == null) {
-            // handling if it ever gets deleted when not supposed to
-            currentOverlay = new DialogueBox(hints);
-        } else {
-            currentOverlay.showDialogueBox(text);
+        if (currentEntity != null) {
+            AnimationRenderComponent animator =  currentEntity.getComponent(AnimationRenderComponent.class);
```
Comment (LucasRyall) [inline source/core/src/main/com/csse3200/game/services/DialogueBoxService.java:78]: You don't need to grab the animation render component, you can just call currentEntity.getEvents().trigger(<animationName>) to start the animation.

**Review by LucasRyall — APPROVED** [review]
Looks good.

**Comment by EllaBerglas** [comment]
I tried to review this but I cannot seem to run it. 
image

**Comment by BaileyS03** [comment]
> I tried to review this but I cannot seem to run it. image

Ill have a look now

**Review by mitsudz — APPROVED** [review]
Looks fine, should probably implement the one simplification Lucas mentioned but doesn't look too important.

**Pull request merged**